tools/analysis_tools/analyze_logs.py

```diff
@@ -73,7 +71,7 @@ def plot_curve(log_dicts, args):
                 for epoch in epochs:
                     iters = log_dict[epoch]['iter']
                     if log_dict[epoch]['mode'][-1] == 'val':
-                        iters = iters[:-1]
+                        iters = iters[:-1] or iters
                     xs.append(
```

Comment:
Using `or iters` is not a good solution because this branch is to handle training logs, it's better to raise an error and guide users to handle this problem.
```suggestion
                        iters = iters[:-1]
                    assert len(iters) > 0, (
                        'The training log is empty, please try to reduce the '
                        'interval of log in config file.')
                    xs.append(
```
Of course, we will refactor this tool to make it more robust to handle different targets.